Add naming convention

skjnldsv opened the pull request:
## Naming convention
To stay consistent, we encourage you to use the following syntax when declaring events

`app-id:action:verb`

### Examples:
- nextcloud:unified-search:close
- contacts:contact:delete
- calendar:event:create
- forms:answer:update

ChristophWurst commented:
What about *before* and *after? Will the after event use the past tense?

nickvergessen commented on README.md:
```diff
@@ -29,3 +29,14 @@ emit('a', {
+## Naming convention
+To stay consistent, we encourage you to use the following syntax when declaring events
+
+`app-id:action:verb`
```
```suggestion
`app-id:object:verb`
```
Event, contact and "unified search" don't sound like action to me :P

skjnldsv commented:
> What about _before_ and *after? Will the after event use the past tense?

Can you provide example please? Feel free to edit this pr

ChristophWurst commented:
> Can you provide example please? Feel free to edit this pr

An event before a file is uploaded vs an event after the file was uploaded. How would you distinguish them?

I personally dislike things like `user:update` when there is no clear semantic about *when* this is fired. Will it call my listener before the change is written? will it call it after? In the worst case ever event does it differently. Then you always have to check the code that emits the event. Let's try to avoid that.

I tried to clarify that for php events at https://docs.nextcloud.com/server/stable/developer_manual/basics/events.html#naming-scheme

skjnldsv commented:
> I tried to clarify that for php events at [docs.nextcloud.com/server/stable/developer_manual/basics/events.html#naming-scheme](https://docs.nextcloud.com/server/stable/developer_manual/basics/events.html#naming-scheme)

"Events are ususally evmitted after the event has happened"

juliushaertl commented:
> I personally dislike things like user:update when there is no clear semantic about when this is fired. Will it call my listener before the change is written? will it call it after? In the worst case ever event does it differently. Then you always have to check the code that emits the event. Let's try to avoid that.

I agree with that. Similar to PHP events I'd rather go for using past tense and a before prefix for others. Of course we can state that "Events are ususally evmitted after the event has happened" but having this represented in the actual event name is way more implicit for reading code.

skjnldsv commented:
Sure!

skjnldsv commented:
@juliushaertl @ChristophWurst 
good enough?

```md
### Examples:
- nextcloud:unified-search:closed
- files:file:uploading
- files:file:uploaded
- contacts:contact:deleted
- calendar:event:created
- forms:answer:updated
```
